ext/RMagick/rmimage.c

```diff
@@ -144,8 +148,11 @@ adaptive_method(int argc, VALUE *argv, VALUE self
  * @return a new image
  */
 static VALUE
-adaptive_channel_method(int argc, VALUE *argv, VALUE self
-                        , Image *fp(const Image *, const ChannelType, const double, const double, ExceptionInfo *))
+#if defined(IMAGEMAGICK_7)
+adaptive_channel_method(int argc, VALUE *argv, VALUE self, Image *fp(const Image *, const double, const double, ExceptionInfo *))
```

Comment:
It is better to use `#if ~ #else ~ # endif` only where it is needed.

```c
#if defined(IMAGEMAGICK_7)
    typedef Image *(channel_method_t)(const Image *, const double, const double, ExceptionInfo *);
#else
    typedef Image *(channel_method_t)(const Image *, const ChannelType, const double, const double, ExceptionInfo *);
#endif

...

static VALUE
adaptive_channel_method(int argc, VALUE *argv, VALUE self, channel_method_t fp)
{
...
```

Reply:
Good idea